src/tpm.rs

```diff
@@ -29,30 +29,510 @@ const RETRY: usize = 4;
+
+    set_tpm_metadata_content("aik", &pem)?;
+    set_tpm_metadata_content("aik_pw", &aik_pw)?;
+    set_tpm_metadata_content("aik_name", load_key_name)?;
+    set_tpm_metadata_content("aik_handle", ak_persistent_handle)?;
+    Ok(())
+}
+
+/*
+ *  Input:
+ *      new password to be set
+ *
+ *  Return:
+ *      Error for unsuccessful password reset
+ *
+ *  Reset the password to the given new password, if new password is generate
+ *  generate a 20 work size length random password from crypto libraray and
+ *  use as new password. Store the final password to tpmdata.json file.
+ */
+fn tpm_set_password(config_pw: String) -> Result<(), KeylimeTpmError> {
+    let owner_pw = match config_pw.as_str() {
+        "generate" => random_password(20)?,
+        _ => config_pw,
+    };
+
+    // Change authenticate, if the first fail, it will do it a second time.
+    if let Err(e) = run(
+        format!("tpm2_changeauth -o {} -e {}", owner_pw, owner_pw),
+        None,
+    ) {
+        run(
+            format!(
+                "tpm2_changeauth -o {} -e {} -O {} -E {}",
+                owner_pw, owner_pw, owner_pw, owner_pw
+            ),
+            None,
+        )?;
+    }
+    set_tpm_metadata_content("owner_pw", &owner_pw)?;
+    Ok(())
+}
+
+/*
+ *  Input:
+ *      hash algorithm for create new key
+ *  Return:
+ *      Error if create new key process failed.
+ *
+ *  Create a new ek key using TPM. Store the ek_handle, ek_password and
+ *  ek_public_key to tpmdata.json file.
+ */
+fn tpm_create_ek(
+    mut asym_alg: Option<String>,
+) -> Result<(), KeylimeTpmError> {
+    let ek_asym_alg = asym_alg.unwrap_or_else(|| String::from("encrypt"));
+    let curr_handle = get_tpm_metadata_content("ek_handle")?;
+    let mut owner_pw = get_tpm_metadata_content("owner_pw")?;
+
+    if curr_handle.is_empty() && owner_pw.is_empty() {
+        let (ret_out, _) =
+            run(format!("tpm_getcap -c handles-persistent"), None)?;
+        let v: Vec<&str> = ret_out.matches(&curr_handle).collect();
+        if !v.is_empty() {
+            run(
+                format!(
+                    "tpm2_evictcontrol -a o -c {} -P {}",
+                    hex::encode(curr_handle),
+                    owner_pw
+                ),
+                None,
+            )?;
+        }
+    }
+
+    if owner_pw.is_empty() {
+        owner_pw = random_password(20)?;
+        set_tpm_metadata_content("owner_pw", &owner_pw);
+    }
+
+    let ek_pw = random_password(20)?;
+    let tf = NamedTempFile::new()?;
+    let tf_path = temp_file_get_path(&tf)?;
+    let paths: Vec<&str> = vec![tf_path];
+    let (ret_out, f_out) = run(
+        format!(
+            "tpm2_createek -c -G {} -p {} -P {} -o {} -e {}",
+            ek_asym_alg, tf_path, ek_pw, owner_pw, owner_pw
+        ),
+        Some(paths),
+    )?;
+
+    let ek_tpm = f_out.get(tf_path).ok_or_else(|| {
+        KeylimeTpmError::new_tpm_rust_error(
+            "Createek output to file content is missing",
```

Comment:
Could we maybe include `tf_path` in the error message to help the user debug? I would rephrase as well:

`Createek file output to file $tf_path cannot be found`